### DISCUSSION TITLE
How to Read a LoRa Packet at The Transmitter end?

### DISCUSSION THREAD

**Discussion by amitdogra25** [sandeepmistry/arduino-LoRa#508] [Q&A] [answered]
Hi, I've been using SX1278 modules successfully for quite some time. The Transmitter has an Arduino Nano and Receiver is NodeMCU. Suddenly the Packet Size at receiver end is appearing as Zero. I rechecked all wiring and everything seems to be fine. Here's the code in setup(): 

```
Serial.begin(115200);
while (!Serial);
Serial.println("LoRa Sender");
LoRa.setPins(ss, rst, dio0);
if (!LoRa.begin(433E6)) {
Serial.println("Starting LoRa failed!");
delay(100);
```

This is the code snippet from Transmitter:
```
LoRa.beginPacket();
LoRa.println(distance);
LoRa.endPacket();
.....
.....
```
Its quite simple. I'm sending some distance data. I'm checking at the Receiver end with this code:
```
int packetSize = LoRa.parsePacket();
if (packetSize) 
  {
    while (LoRa.available())
    {
      LoRaData = LoRa.readString();
        Serial.println(LoRaData);
.....
.....
```
Suddenly packetSize is coming Zero. I don't know if anything is really being sent from the Transmitter. 
Is there a way to check the packet at Transmitter end just after sending so I'm sure that indeed something is being sent. If Sending is OK then I can troubleshoot the Receiver end. Any guidance is much appreciated. Thanks.

**Comment by IoTThinks** [comment, marked as answer]
Try receiver callback example instead of parsePacket.

ParsePacket is like hearing for short period then stop.

So if no sending packet during ParsePacket then the size is 0. It is normal.

**Reply by amitdogra25** [reply]
Thanks @IoTThinks 
I tried the Receiver Callback example and I can see some improvement. Every few seconds some jumbled characters are appearing which is encouraging. While I continue my investigation further, I am wondering if it's possible for a perfectly working Transmitter-Receiver pair of SX1278 to suddenly start behaving like this. Earlier I was able to transmit data every second, receive it in perfect shape and all was fine. Any thoughts?

**Reply by IoTThinks** [reply]
Last time is by luck only.

When the parsePacket has less time to run due to longer loop() or your two nodes are eventualy out of sync.

**Reply by amitdogra25** [reply]
I understand. Thanks for the explanation. I've got a lot to learn.
